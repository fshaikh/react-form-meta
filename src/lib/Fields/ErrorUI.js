import React from 'react';
import '../Form.css';

export default function ErrorUI({showError, errorMessages}) {
    if(errorMessages.length === 0) {
        return '';
    }
    var index = 0;
    return (
        <ul className="validation-error">
            {errorMessages.map((message) => {
                return <li key={index++}>
                    <span className="validation-error-text">{message}</span>
                </li>
            })}
            
        </ul>
    )
    
}