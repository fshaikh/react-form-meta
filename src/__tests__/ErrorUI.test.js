import React from 'react';
import ErrorUI from '../lib/Fields/ErrorUI';
import renderer from 'react-test-renderer';

describe('Error UI',() => {
    describe('Snapshot tests for Error UI', () => {
        it('should render empty when showEror is false', () => {
            var showError = false;
            const errorMessages =[];
            const tree = renderer.create(<ErrorUI showError={showError} errorMessages={errorMessages} />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('should render empty when showEror is true and no error messages', () => {
            var showError = true;
            const errorMessages =[];
            const tree = renderer.create(<ErrorUI showError={showError} errorMessages={errorMessages} />).toJSON();
            expect(tree).toMatchSnapshot();
        });
        
        it('should render error messages ', () => {
            var showError = true;
            const errorMessages =['Error message 1','Error message 2'];
            const tree = renderer.create(<ErrorUI showError={showError} errorMessages={errorMessages} />).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
   
});