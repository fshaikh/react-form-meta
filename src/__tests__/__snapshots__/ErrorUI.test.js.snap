// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error UI Snapshot tests for Error UI should render empty when showEror is false 1`] = `""`;

exports[`Error UI Snapshot tests for Error UI should render empty when showEror is true and no error messages 1`] = `""`;

exports[`Error UI Snapshot tests for Error UI should render error messages  1`] = `
<ul
  className="validation-error"
>
  <li>
    <span
      className="validation-error-text"
    >
      Error message 1
    </span>
  </li>
  <li>
    <span
      className="validation-error-text"
    >
      Error message 2
    </span>
  </li>
</ul>
`;
